server/app/assets/javascripts/admin_programs.ts

```diff
@@ -211,15 +230,28 @@ class AdminPrograms {
    * @param {boolean} shouldHide - Whether to show or hide the required indicator
    */
   static hideRequiredIndicators(fieldSelector: string, shouldHide: boolean) {
-    const field = document.querySelector(fieldSelector)
-    const requiredIndicators = field?.querySelectorAll('span')
-    requiredIndicators?.forEach((indicator) => {
-      if (shouldHide) {
-        indicator.classList.add('hidden')
-      } else {
-        indicator.classList.remove('hidden')
-      }
-    })
+    const labelElement = document.querySelector(fieldSelector)
+    if (!labelElement) {
+      return
+    }
+
+    let requiredSpan = labelElement.querySelector(
+      'span.text-red-600.font-semibold',
+    )
+    if (!requiredSpan) {
```

Comment:
praise/suggestion: great idea adding in the required spans when they're missing. i have two suggestions:
1. move this into it's own method for adding required spans that is not nested i this method since it's a separate concern. or change the name of this method to something more general like `handleRequiredIndicators`.
2. this is currently resulting in a funky state for application steps 
<img width="234" alt="image" src="https://github.com/user-attachments/assets/7749d6bc-7860-42d5-9a22-0eba7b90bc2c" />

you could remove the "optional" label from step 1, or (my preference) make this change in a separate PR since it's a broader concern than just for external URLS.